File: app/src/main/java/com/fieldservice/app/utils/DateFormatter.java
package com.fieldservice.app.utils;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/** Centraliza o formato de data exibido na UI, evitando padrões repetidos pelas telas. */
public final class DateFormatter {

    private static final SimpleDateFormat DISPLAY_FORMAT =
            new SimpleDateFormat("dd/MM/yyyy", new Locale("pt", "BR"));

    private DateFormatter() {
    }

    public static String format(Date date) {
        return DISPLAY_FORMAT.format(date);
    }
}

File: legacy-kotlin/app/src/main/java/com/fieldservice/app/utils/DateFormatter.kt
package com.fieldservice.app.utils

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/** Centraliza o formato de data exibido na UI, evitando padrões repetidos pelas telas. */
object DateFormatter {
    // Locale.of() (substituto não-depreciado) exige API 31+; o construtor abaixo
    // é o único caminho compatível com o minSdk 24 deste projeto.
    @Suppress("DEPRECATION")
    private val displayFormat = SimpleDateFormat("dd/MM/yyyy", Locale("pt", "BR"))

    fun format(date: Date): String = displayFormat.format(date)
}
